apps/smart-contracts/core/helpers/index.ts

```diff
@@ -26,3 +34,91 @@ export async function createMarket(marketParams: CreateMarketParams): Promise<Cr
     hash: events[0].args.longShortHash,
   }
 }
+
+export async function getBaseTokenAmount(
+  collateral: Collateral | MockContract<Collateral>,
+  collateralAmount: BigNumber
+): Promise<BigNumber> {
+  const factorAfterFee = BigNumber.from(FEE_DENOMINATOR).sub(await collateral.getDepositFee())
+  return collateralAmount.mul(factorAfterFee).div(FEE_DENOMINATOR)
```

Comment:
this still looks incorrect
it should be:
```suggestion
  const afterFeeFactor = (BigNumber.from(FEE_DENOMINATOR).sub(await collateral.getDepositFee()).div(FEE_DENOMINATOR)
  return collateralAmount.div(afterFeeFactor)
```

Reply:
Ah sorry, I was writing this as if we were writing `getCollateralAmount`

Reply:
I have fixed this so that it correctly takes in the `collateralAmount` that someone wants to mint, and calculates the amount of base token needed. So the above wouldn't be correct either, but I have modified it. Also keep in mind it seems like you do not like doing a `.mul(X).div(Y)` as the final return value, and prefer a singular `.mul` or `.div` with a factor.

Understand that we can't do that because we are performing `BigNumber` integer math, and are doing so to ensure the value returned is the same as what would be returned within the contract. `afterFeeFactor` in the above code would return 0 every time.